application.go

```diff
@@ -109,6 +129,28 @@ type DeleteAppOpts struct {
 	Force bool `url:"force,omitempty"`
 }
 
+// Diego ======
+
+// SetIPAddressPerTask defines that the application will have a IP address defines by a external agent. this
+// configuration is not allowed to be used with Port or PortDefinitions.
+func (r *Application) SetIPAddressPerTask(ipAddressPerTask *IPAddressPerTask) *Application {
+
+	r.Ports = make([]int, 0)
+	r.EmptyPortDefinitions()
+	r.IPAddressPerTask = ipAddressPerTask
+
+	return r
+}
+
+// EmptyIPAddressPerTask explicitly empties port definitions -- use this if you need to empty
+// IP address per task of an application that already has some definitions set (setting port definitions to nil will
+// keep the current value)
+func (r *Application) EmptyIPAddressPerTask() *Application {
+	r.IPAddressPerTask = nil
```

Comment:
I did some tests in this feature and it's exactly what you said, once enabled IP-per-task it is impossible to disable. I'll remove the method.